executor/src/witgen/block_processor.rs

```diff
@@ -28,12 +30,21 @@ impl<'a, 'b, 'c, T: FieldElement, Q: QueryCallback<T>> BlockProcessor<'a, 'b, 'c
     pub fn new(
         row_offset: RowIndex,
         data: FinalizableData<T>,
+        publics: BTreeMap<&'a str, T>,
```

Comment:
Thinking about it: No, because these are the already known values for the publics, right?

Maybe this can be combined with the values for challenges in the future...

Reply:
I grouped the trace block and publics, using the same type returned by `finish()`.